pinot-segment-local/pom.xml

```diff
@@ -156,5 +156,30 @@
       <type>test-jar</type>
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.carrotsearch</groupId>
+      <artifactId>hppc</artifactId>
```

Comment:
We use `it.unimi.dsi.fastutil` for the primitive type data structures (e.g. `it.unimi.dsi.fastutil.ints.Int2IntOpenHashMap`). Let's try to reduce the external dependency unless there is some clear benefits of introducing this new library